tuned_lens/scripts/train_loop.py

```diff
@@ -16,246 +18,362 @@
+
+    model: Model
+    """Model configuration."""
+
+    data: Data
+    """Data configuration."""
+
+    sharding: Sharding
+    """Sharding configuration."""
+
+    seed: int = 42
+    """Random seed for data shuffling."""
+
+    lens_name_or_path: Optional[str] = field(alias=["-l"], default=None)
+
+    constant: Optional[bool] = field(action="store_true")
+    """Train only the bias term."""
+
+    lasso: float = 0.0
+    """LASSO (L1) regularization strength."""
+
+    lr_scale: float = 1.0
+    """The default LR (1e-3 for Adam, 1.0 for SGD) is scaled by this factor."""
+
+    momentum: float = 0.9
+    """Momentum coefficient for SGD, or beta1 for Adam."""
+
+    num_steps: int = 250
+    """Number of training steps."""
+
+    optimizer: OptimizerOption = OptimizerOption.SGD
+    """The type of optimizer to use."""
+
+    output: Optional[Path] = field(alias=["-o"], default=None)
+    """File to save the lenses to. Defaults to the model name."""
+
+    pre_ln: Optional[bool] = field(action="store_true")
+    """Apply layer norm before, and not after, each probe."""
+
+    resume: Optional[Path] = None
+    """File to resume training from."""
+
+    separate_unembeddings: Optional[bool] = field(action="store_true")
+    """Learn a separate unembedding for each layer."""
+
+    tokens_per_step: int = 2**18
+    """Number of tokens per step."""
+
+    wandb: Optional[str] = None
+    """Name of run in Weights & Biases."""
+
+    warmup_steps: Optional[int] = None
+    """Number of warmup steps. Defaults to min(0.1 * num_steps, 1000) for Adam and 0
+    for SGD."""
+
+    weight_decay: float = 1e-3
+    """Weight decay coefficient."""
+
+    zero: Optional[bool] = field(action="store_true")
+    """Use ZeroRedundancyOptimizer."""
+
+    token_shift: Optional[int] = None
+    """How to shift the labels wrt the input tokens (1 = next token, 0 = current token,
+    -1 = previous token, etc.)"""
+
+    per_gpu_batch_size: int = 1
+    """Number of samples to try to fit on a GPU at once."""
+
+    loss: LossChoice = LossChoice.KL
+    """Loss function to use."""
+
+    def get_lens(self, model: PreTrainedModel) -> TunedLens:
+        """Load or create a TunedLens model."""
+        if self.lens_name_or_path is None:
+            lens = TunedLens(model)
+        else:
+            lens = TunedLens.load(self.lens_name_or_path)
+
+        lens_size = sum(p.numel() * p.element_size() for p in lens.parameters())
+        print(f"Tuned lens memory usage: {lens_size / 2 ** 20:.2f} MB per GPU")
+
+        if self.constant:
+            for probe in lens:
+                probe.weight.requires_grad_(False)
+
+        return lens
+
+    def _init_logging(self, model_name: str, lens: TunedLens):
+        if local_rank() > 0 or not self.wandb:
+            return
+
         import wandb
 
-        *_, name = args.model_name.split("/")
         wandb.init(
-            config=vars(args),
+            config=vars(self),
             entity="eleutherai",
```

Comment:
We do need to make this customizable or something

Reply:
Ya this definitly needs to get fixed.